templates/writing-task-topics.rst

```diff
@@ -0,0 +1,240 @@
+
+
+.. _task-topics-intro:
+
+Introduction
+============
+
+This section will consist of the below, all of which need to be
+written.
+
+- **Summary/context (1-2 sentences):**
+
+.. Note:: Write this section concisely, as it should be very brief,
+          and is primarily about whether the reader should choose to
+          use this task to achieve h/her goal in a very quick scan.
+
+For example, for `ProcessCcd` we might write:
+
+.. code-block:: rst
+
+ ProcessCcdTask is a command line task which executes the processing
+ steps to turn raw pixel-level data into characterized images and
+ calibrated catalogs.
+	  
+	  
+- **Concise summary of logic/algorithm in a paragaph and/or bullet list.**
+
+.. Note:: Be concise and link to other tasks wherever needed.  This
+          can be a few more sentences, but should not contain very
+          many details or math at this point (that will go in the
+          algorithmic section at the bottom).  It should just say
+          where this fits into the larger DM structure, what
+          retargetable subtasks or methods within a task it calls by
+          default.
+
+For example, for `ProcessCcd` we would give a list of the 3 subtasks
+that `ProcessCcd` invokes to do its job.
+	  
+.. Note:: If there are optional tasks that are called you can choose
+          to fill those in here as well, but note them as optional and
+          depending on whether a flag is set in the configuration
+          parameters
+
+.. _task-topics-module:
+
+- **Module membership:**
+
+.. Note::  Here, simply state the module the task is implemented inside of, one sentence is sufficient.
+	   
+.. _task-topics-seealso:
+	   
+- **"SeeAlso" box:**
+
+.. Note:: The API Usage page for this task will automatically be
+          linked to in this box, but please fill in the following
+          types of tasks and pages into this section:
+
+  - Tasks that commonly use this task
+  
+  - Tasks that can be used instead of this task
+
+  - Pages in the Processing and Frameworks sections of the Science Pipelines documentation.
+
+In the case of `ProcessCcd`, we would simply say is called as a
+`command line task`, where for e.g. IsrTask, we would say that that it
+is most commonly called by `ProcessCcd`.
+
+.. _task-topics-config:	  
+    
+Configuration
+=============
+
+.. _task-topics-retarg:
+
+Retargetable Subtasks
+---------------------
+
+.. Note:: This content will be automatically derived from the Task
+          code (in which keywords that describe the subtasks in the
+          python code, including a `doc` keyword in which the subtask
+          is described, will already be written).
+
+.. _task-topics-params:
+   
+Parameters
+----------
+
+
+.. Note:: Like the Retargetable Subtasks, this content will be derived
+          from keywords that describe the parameters where they are
+          defined in the Task Config class.
+
+
+.. _task-topics-python:
+   
+Python usage
+============
+
+.. _task-topics-initzn:
```

Comment:
Yeah, sorry - i by default use abbreviations that are very familiar to me in links etc., forgetting at times this is going to be embedded in larger structures and every word will be open to public scrutiny.  Will try to be more aware of this in the future.

Changed.